client/my-sites/checklist/checklist-show/index.jsx

```diff
@@ -132,15 +133,17 @@ class ChecklistShow extends PureComponent {
 	}
 
 	render() {
-		const { displayMode, siteId, tasks } = this.props;
+		const { siteId, tasks } = this.props;
+		const parsedUrl = url.parse( location.href, true );
+		const displayMode = parsedUrl.query.d;
```

Comment:
I'm not sure the component should be aware where `displayMode` comes from, or how to parse the `url`. I'd rather keep this on the controller and pass down a `prop`.
The controller already provides access to query string parameters via `context.query`.

Reply:
Yep - thanks! that makes sense - i'll change it

Reply:
As a general note, try to be careful with with unguarded property accessors`parsedUrl.query.d`. It's often the case where one of the objects on the chain may be null or undefined (esp if we're dealing with global state).